rust_dev_preview/cross_service/rest_ses/src/work_item/repository.rs

```diff
@@ -0,0 +1,205 @@
+    let statement = match archive {
+        WorkItemArchived::All => statement.sql(format!(r#"SELECT {FIELDS} FROM Work;"#)),
+        _ => statement
+            .sql(format!(
+                r#"SELECT {FIELDS} FROM Work WHERE archive = :archive ;"#
+            ))
+            .set_parameters(params![("archive", format!("{}", u8::from(&archive)))]),
+    };
+
+    let statement = statement
+        .format_records_as(RecordsFormatType::Json)
+        .send()
+        .await;
+
+    parse_rds_output(statement)
+}
+
+/// Update a single item in the database, by ID.
+/// Retrieves the value after update for its result.
+#[tracing::instrument(name = "Repository Update WorkItem", skip(client))]
+pub async fn update(item: &WorkItem, client: &RdsClient) -> Result<WorkItem, WorkItemError> {
+    client
+        .execute_statement()
+        .sql(
+            r#"
+            UPDATE Work
+            SET 
+                username = :username,
+                date = :date,
+                description = :description,
+                guide = :guide,
+                status = :status,
+                archive = :archive
+            WEHRE idwork = :idwork;
+        "#,
+        )
+        .set_parameters(params![
+            ("idwork", item.idwork),
+            ("username", item.name),
+            ("date", format!("{}", item.date.format(RDS_DATE_FORMAT))),
+            ("description", item.description),
+            ("guide", item.guide),
+            ("status", item.status),
+            ("archive", format!("{}", u8::from(&item.archive)))
+        ])
+        .send()
+        .await
+        .map_err(|err| {
+            tracing::error!("Failed to update user: {id} {err:?}", id = item.idwork);
+            WorkItemError::RDSError(err.into())
+        })?;
+    retrieve(item.idwork().to_string(), client).await
+}
+
+/// Delete an item from the database, returning () on success.
+#[tracing::instrument(name = "Repository Delete WorkItem", skip(client))]
+pub async fn delete(id: String, client: &RdsClient) -> Result<(), WorkItemError> {
+    client
+        .execute_statement()
+        .sql(r#"DELETE FROM Work WHERE idwork = :idwork"#)
+        .set_parameters(params![("idwork", id)])
+        .send()
+        .await
+        .map(|_| Ok(()))
+        .map_err(|err| {
+            tracing::error!("Failed to delete user: {id} {err:?}");
+            WorkItemError::RDSError(err.into())
+        })?
+}
+
+/// Attempt to parse an RDS Data SQL statement to a WorkItem.
+/// This relies on formatting the SQL statement response in a way that matches serde annotations in WorkItem.
+fn parse_rds_output(
+    statement: Result<ExecuteStatementOutput, SdkError<ExecuteStatementError>>,
+) -> Result<Vec<WorkItem>, WorkItemError> {
+    // Did the request succeed? It could fail because the SQL is incorrect or the IAM permissions
+    // are wrong, because of a failing network connection, or because of a problem in AWS itself.
```

Comment:
// are wrong. Alternatively, it could be because of a failing network connection or a problem in AWS itself.